wordpress-mobile/WordPress-Android
# Jetpack Mobile: Scan Introduce UiState

Parent #13326

This is a small preparatory PR with a scope to just link things (Service, VM, Fragment, ViewHolder) together for the Scan screen UI. It 
- introduces recycler view + adapter to the `ScanFragment`
- introduces Content UiState to the `ScanViewModel` 
- populates basic elements (scan state icon, title) on the screen using the state received from the scan service. 

I plan to use (in the upcoming PRs) different view types - scan state (with icon, title, description, action buttons), progress, threat header, threat items e.t.c. inside a RecyclerView so that entire screen is scrollable .

**To test**

Prerequisite: Make sure both Scan feature flag and MySiteImprovements flag are set on in the App settings.

- Select a site with scan capability on the My Site tab
- Notice that scan menu is displayed
- Click scan menu item 
- Notice that basic elements (icon, title) are displayed on the Scan screen from the service. 

![device-2020-12-10-170621](https://user-images.githubusercontent.com/1405144/101768107-2ff20000-3b0b-11eb-9e3f-64e217183602.png)

**Note**
- Used placeholder icons, texts. Ignore all stying.
- `ScanListItemState` has to be fine tuned. I'll pickup and fine tune different view types (with their states) in the upcoming PRs.
- I'll create a separate FluxC PR for storing/ retrieving threats into/ from db.

**Merge Instructions**

1. Make sure #13555 is merged
2. Remove "Not ready for merge" label
3. Merge this PR to develop

PR submission checklist:
